contracts/most/Utmost.sol

```diff
@@ -0,0 +1,155 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "../proxies/MasterCopyNonUpgradable.sol";
+import "./GenesisUtmost.sol";
+import "./../consensus/CoconsensusModule.sol";
+import "./../utilityToken/UtilityToken.sol";
+
+/**
+ * @title Utmost contract implements UtilityToken.
+ *
+ * @dev Utmost is an ERC20 token wrapper for the base coin that is used to pay
+ *      for gas consumption on the auxiliary chain.
+ */
+contract Utmost is MasterCopyNonUpgradable, GenesisUtmost, UtilityToken, CoconsensusModule {
+
+    /* Events */
+
+    /** Emitted whenever the ERC20 Utmost is converted to base coin Utmost. */
+    event TokenUnwrapped(
+        address indexed _account,
+        uint256 _amount
+    );
+
+    /** Emitted whenever the base coin Utmost is converted to ERC20 Utmost. */
+    event TokenWrapped(
+        address indexed _account,
+        uint256 _amount
+    );
+
+
+    /** Constants */
+
+    /** Token Symbol */
+    string public constant TOKEN_SYMBOL = "UM";
+
+    /** Token Name */
+    string public constant TOKEN_NAME = "Utmost";
+
+    /** Token Decimal */
+    uint8 public constant TOKEN_DECIMALS = 18;
+
+
+    /* External Functions */
+
+    /**
+     * @notice Calls UtilityToken.setup.
+     *
+     * @dev Function requires:
+     *      - msg.sender should be Coconsensus
+     *
+     * @dev This function must be called only once. This check is done in setup
+     *      function of UtilityToken.
+     */
+    function setup()
+        external
+        onlyCoconsensus
+    {
+        UtilityToken.setup(
+            TOKEN_SYMBOL,
+            TOKEN_NAME,
+            TOKEN_DECIMALS,
+            genesisTotalSupply,
+            address(0x0000000000000000000000000000000000004d02)
+        );
+
+        balances[address(this)] = genesisTotalSupply;
```

Comment:
Add a dev comment to explain why this is needed.

Reply:
Updated!

Reply:
Please explain as an inline comment. Below is an example, please rephrase it properly.
`In the genesis block, the validators gets the base token equivalent to the amount that they deposited before joining the core. So the equivalent amount of ERC20 token should be minted and Utmost contract should hold the balance. This is what should happen in a normal mint process. So in the setup the Utmost token will hold the ERC20 token equal to  genesisTotalSupply`.